content/en/docs/ops/ambient/getting-started/test.sh

```diff
@@ -31,48 +31,92 @@ _wait_for_daemonset istio-system istio-cni-node
 _verify_contains snip_verify_traffic_notsleep_to_productpage "$snip_verify_traffic_notsleep_to_productpage_out"
 
 snip_adding_your_application_to_ambient_addtoambient_1
 
 # test traffic after ambient mode is enabled
-_verify_contains snip_verify_traffic_sleep_to_ingress "$snip_verify_traffic_sleep_to_ingress_out"
+if [ "$GATEWAY_API" == "true" ]; then
+    _verify_contains snip_verify_traffic_sleep_to_gateway "$snip_verify_traffic_sleep_to_gateway_out"
+else
+  _verify_contains snip_verify_traffic_sleep_to_ingress "$snip_verify_traffic_sleep_to_ingress_out"
+fi
+
 _verify_contains snip_verify_traffic_sleep_to_productpage "$snip_verify_traffic_sleep_to_productpage_out"
 _verify_contains snip_verify_traffic_notsleep_to_productpage "$snip_verify_traffic_notsleep_to_productpage_out"
 
-snip_l4_authorization_policy_1
-_verify_contains snip_verify_traffic_sleep_to_ingress "$snip_verify_traffic_sleep_to_ingress_out"
+if [ "$GATEWAY_API" == "true" ]; then
+  snip_l4_authorization_policy_5
+  _verify_contains snip_verify_traffic_sleep_to_gateway "$snip_verify_traffic_sleep_to_gateway_out"
+else
+  snip_l4_authorization_policy_1
+  _verify_contains snip_verify_traffic_sleep_to_ingress "$snip_verify_traffic_sleep_to_ingress_out"
+fi
+
 _verify_contains snip_verify_traffic_sleep_to_productpage "$snip_verify_traffic_sleep_to_productpage_out"
 _verify_contains snip_verify_traffic_notsleep_to_productpage "command terminated with exit code 56"
 
 snip_l7_authorization_policy_1
 snip_l7_authorization_policy_2
-_verify_contains snip_l7_authorization_policy_3 "Deployed waypoint proxy"
+_verify_contains snip_l7_authorization_policy_2 "Deployed waypoint proxy"
+
+if [ "$GATEWAY_API" == "true" ]; then
+  snip_l7_authorization_policy_8
+  _verify_contains snip_l7_authorization_policy_9 "$snip_l7_authorization_policy_9_out"
+else
+  snip_l7_authorization_policy_4
+  _verify_contains snip_l7_authorization_policy_5 "$snip_l7_authorization_policy_5_out"
+fi
 
-snip_l7_authorization_policy_5
 _verify_contains snip_l7_authorization_policy_6 "$snip_l7_authorization_policy_6_out"
 _verify_contains snip_l7_authorization_policy_7 "$snip_l7_authorization_policy_7_out"
-_verify_contains snip_l7_authorization_policy_8 "$snip_l7_authorization_policy_8_out"
 
 snip_control_traffic_control_1
 _verify_contains snip_control_traffic_control_1 "waypoint default/bookinfo-reviews applied"
 
-snip_control_traffic_control_2
-_verify_lines snip_control_traffic_control_3 "
-+ reviews-v1
-+ reviews-v2
-- reviews-v3
-"
+if [ "$GATEWAY_API" == "true" ]; then
+  snip_control_traffic_control_6
+  snip_control_traffic_control_7
+  _verify_lines snip_control_traffic_control_8 "
+  + reviews-v1
+  + reviews-v2
+  - reviews-v3
+  "
+else
+  snip_control_traffic_control_2
+  _verify_lines snip_control_traffic_control_3 "
+  + reviews-v1
+  + reviews-v2
+  - reviews-v3
+  "
+fi
 
 # @cleanup
 cleanup_bookinfo_sample
-snip_uninstall_uninstall_1
+if [ "$GATEWAY_API" == "true" ]; then
```

Comment:
Cleanup instructions are special. The gateway-api instructions need to be moved to gtwapi_test.sh, which will need to be added. Also, you need to define GATEWAY_API at the top of test.sh (see other doc tests, for example).

Reply:
Will do. Thanks for all the feedback and guidance.